src/backend/api/backend.did

```diff
@@ -247,6 +247,56 @@ type UpdateProposalReviewResponse = variant {
   err : Err;
 };
 
+type ReviewCommitState = variant {
+  reviewed : record {
+    matches_description : bool;
+    comment : opt text;
+    highlights : vec text;
+  };
+  not_reviewed;
+};
+
+type ProposalReviewCommit = record {
+  proposal_review_id : text;
+  user_id : text;
+  created_at : text;
+  commit_sha : text;
+  state : ReviewCommitState;
+};
+
+type CreateProposalReviewCommitRequest = record {
```

Comment:
It would be nice to have these properties optional so the UI can autosave as soon as there is any available information, rather than waiting until all information is available.

Reply:
If we remove the commit_sha from the index, the field here can be optional.
Regarding the state, we can make it optional and default it to `not_reviewed` in the backend.